sdk/appconfiguration/app-configuration/src/internal/synctokenpolicy.ts

```diff
@@ -22,34 +20,21 @@ export const SyncTokenHeaderName = "sync-token";
  * @param syncTokens - the sync tokens store to be used across requests.
  * @internal
  */
-export function syncTokenPolicy(syncTokens: SyncTokens): RequestPolicyFactory {
+export function syncTokenPolicy(syncTokens: SyncTokens): PipelinePolicy {
   return {
-    create: (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => {
-      return new SyncTokenPolicy(nextPolicy, options, syncTokens);
-    },
-  };
-}
-
-class SyncTokenPolicy extends BaseRequestPolicy {
-  constructor(
-    nextPolicy: RequestPolicy,
-    options: RequestPolicyOptions,
-    private _syncTokens: SyncTokens
-  ) {
-    super(nextPolicy, options);
-  }
-
-  public async sendRequest(webResource: WebResource): Promise<HttpOperationResponse> {
-    const syncTokenHeaderValue = this._syncTokens.getSyncTokenHeaderValue();
+    name: "Sync Token Policy",
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      const syncTokenHeaderValue = syncTokens.getSyncTokenHeaderValue();
 
-    if (syncTokenHeaderValue != null) {
-      webResource.headers.set(SyncTokenHeaderName, syncTokenHeaderValue);
-    }
+      if (syncTokenHeaderValue != null) {
```

Comment:
this should always be truthy right? empty string and 0 aren't going to be helpful either?
```suggestion
      if (syncTokenHeaderValue) {
```